src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/scheduler/AmendGeneralEmailTask.java

```diff
@@ -0,0 +1,163 @@
+
+        log.info("CaseReferences has {} cases.", caseReferences.size());
+        return caseReferences;
+    }
+
+    @Override
+    protected String getCaseListFileName() {
+        return "caserefs-encrypted.csv";
+    }
+
+    @Override
+    protected String getTaskName() {
+        return TASK_NAME;
+    }
+
+    @Override
+    protected boolean isTaskEnabled() {
+        return taskEnabled;
+    }
+
+    @Override
+    protected CaseType getCaseType() {
+        return CaseType.forValue(caseTypeId);
+    }
+
+    @Override
+    protected String getSummary() {
+        return SUMMARY;
+    }
+
+    @Override
+    protected void executeTask(FinremCaseDetails finremCaseDetails) {
+        FinremCaseData caseData = finremCaseDetails.getData();
+        ObjectMapper mapper = new ObjectMapper();
+
+        if (caseData.getGeneralEmailWrapper() != null
+            && caseData.getGeneralEmailWrapper().getGeneralEmailUploadedDocument() != null) {
+            try {
+                log.info("Case {} GeneralEmailUploadedDocument before: {}", finremCaseDetails.getId(), mapper
+                        .writerWithDefaultPrettyPrinter()
+                        .writeValueAsString(caseData.getGeneralEmailWrapper().getGeneralEmailUploadedDocument()));
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+
+            caseData.getGeneralEmailWrapper().setGeneralEmailUploadedDocument(null);
+            log.info("Case {} generalEmailUploadedDocument set to null", finremCaseDetails.getId());
+
+        } else {
+            log.info("Case {} has empty generalEmailUploadedDocument field", finremCaseDetails.getId());
+        }
+    }
+
+    void setSecret(String secret) {
+        this.secret = secret;
+    }
+
+    void setTaskEnabled(boolean taskEnabled) {
+        this.taskEnabled = taskEnabled;
+    }
+
+    void setCaseTypeContested() {
+        this.caseTypeId = CaseType.CONTESTED.getCcdType();
+    }
+
+    public static void encryptFile(String inputFilePath, String outputFilePath, SecretKey key) throws Exception {
+        String content = new String(Files.readAllBytes(Paths.get(inputFilePath)));
+        CaseReferenceCsvLoader csvLoader = new CaseReferenceCsvLoader();
+        String encryptedContent = csvLoader.encrypt(content, key);
+        Files.write(Paths.get(outputFilePath), encryptedContent.getBytes());
+    }
+
+    public static void decryptFile(String inputFilePath, String outputFilePath, SecretKey key) throws Exception {
+        String encryptedContent = new String(Files.readAllBytes(Paths.get(inputFilePath)));
+        CaseReferenceCsvLoader csvLoader = new CaseReferenceCsvLoader();
+        String decryptedContent = csvLoader.decrypt(encryptedContent, key);
+        Files.write(Paths.get(outputFilePath), decryptedContent.getBytes());
+    }
+
+    public static void main(String[] args) throws Exception {
```

Comment:
This code should be in a separate class. The code doesn't belong in this class. The need to encrypt a file is going to be required by all cron jobs that use a CSV file and not just amending general emails